src/Pate/Config.hs

```diff
@@ -202,6 +204,16 @@
   | ContinueAfterFailure
   | ContinueAfterRecoverableFailures
 
+-- | Controls how abstract domains are shared between function call nodes in the pairgraph.
+data ContextSensitivity =
+    -- | All calls to any given function share the same abstract domain, i.e.
+    --   the resulting domain is proven to hold at every return site.
+    AllSharedAbstractDomains
+    -- | Each call to a function has a distinct abstract domain that holds
+    --   at its (single) return site (distinguished by tagging the function node with
+    --   the calling node)
+  | AllDistinctAbstractDomains
```

Comment:
"All" is not quite correct, is it? What is the key in the map from contexts to abstract values? What happens in the presence of recursion?